types/types.go

```diff
@@ -3,10 +3,17 @@
 
 package types
 
-import "errors"
+import (
+	"errors"
+
+	"github.com/hashicorp/raft"
+)
 
 var (
-	ErrNotFound = errors.New("log entry not found")
+	// ErrNotFound The raft library checks equality on the error to handle
+	// certain cases, so we must return the same exact error as the raft
+	// library defined here.
```

Comment:
```suggestion
	// ErrNotFound is returned when there is no log at the requested index. 
	// The raft library checks equality on the error to handle
	// certain cases, so we must return the same exact error as the raft
	// library defined here.
```

Reply:
nit: Go comments should read naturally. Apologies I didn't make that clearer in my previous suggestion!